test/problematic_tests/impl_test.py

```diff
@@ -0,0 +1,52 @@
+import os
+import pytest
+import unittest
+
+from GenomeFileUtil.GenomeFileUtilImpl import GenomeFileUtil
+from conftest import assert_exception_correct
+
+
+class ImplTest(unittest.TestCase):
+    def setUp(self):
+        self.cfg = {}
+        self.cfg["workspace-url"] = "https://ci.kbase.us/services/ws"
+        self.cfg["shock-url"] = "https://ci.kbase.us/services/shock-api"
+        self.cfg["handle-service-url"] = "https://ci.kbase.us/services/handle_service"
+        self.cfg["scratch"] = "/kb/module/work/tmp"
+        self.cfg["srv-wiz-url"] = "https://ci.kbase.us/services/service_wizard"
+        self.cfg["auth-service-url"] = "https://ci.kbase.us/services/auth/api/legacy/KBase/Sessions/Login"
+        self.cfg["re-api-url"] = "https://ci.kbase.us/services/relation_engine_api"
+
+    def tearDown(self):
+        for kbase_secure_param in (
+            "KBASE_SECURE_CONFIG_PARAM_MAX_THREADS",
+            "KBASE_SECURE_CONFIG_PARAM_THREADS_PER_CPU",
+        ):
+            os.environ.pop(kbase_secure_param, None)
+
+    def _run_test_fail(self, cfg, error_message):
+        with pytest.raises(Exception) as got:
+            GenomeFileUtil(cfg)
+        assert_exception_correct(got.value, ValueError(error_message))
+
+    def test_valid_catalog_param_type(self):
+        os.environ["KBASE_SECURE_CONFIG_PARAM_MAX_THREADS"] = "10"
```

Comment:
This should be a non default value - as it is the code could break such that the default value is always used and this test would still pass

Reply:
updated!

Reply:
:+1: